server/src/test/java/org/opensearch/action/admin/cluster/snapshots/get/GetSnapshotsActionTests.java

```diff
@@ -0,0 +1,245 @@
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+    }
+
+    public void testCurrentRepoSnapshot() {
+        final Task task = createTask();
+        RepositoryMetadata repositoryMetadata = new RepositoryMetadata(repoName, repoType, Settings.EMPTY);
+        Repository repository = repositoriesService.createRepository(repositoryMetadata, repositoriesService.typesRegistry());
+        ;
+        Map<String, Repository> repositoryMap = new HashMap<>();
+        repositoryMap.put(repoName, repository);
+        repositoriesService = repositoriesService.repositories(repositoryMap);
+        assertEquals(repositoryMap, repositoriesService.repositories());
+
+        GetSnapshotsRequest repoSnapshotRequest = new GetSnapshotsRequest().repository(repoName)
+            .snapshots(currSnapshotOnly)
+            .ignoreUnavailable(true);
+        getSnapshotsAction.execute(null, repoSnapshotRequest, ActionListener.wrap(repoSnapshotResponse -> {
+            assertNotNull("snapshots should be set as we are checking the current snapshot", repoSnapshotResponse.getSnapshots());
+        }, exception -> { throw new AssertionError(exception); }));
+
+        repoSnapshotRequest = new GetSnapshotsRequest().repository(repoName)
+            .snapshots(currSnapshotOnly)
+            .ignoreUnavailable(true)
+            .verbose(false);
+        getSnapshotsAction.execute(null, repoSnapshotRequest, ActionListener.wrap(repoSnapshotResponse -> {
+            assertNotNull("snapshots should be set as we are checking the current snapshot", repoSnapshotResponse.getSnapshots());
+        }, exception -> { throw new AssertionError(exception); }));
+
+        repoSnapshotRequest = new GetSnapshotsRequest().repository(repoName)
+            .snapshots(currSnapshotOnly)
+            .ignoreUnavailable(false)
+            .verbose(false);
+        getSnapshotsAction.execute(null, repoSnapshotRequest, ActionListener.wrap(repoSnapshotResponse -> {
+            assertNotNull("snapshots should be set as we are checking the current snapshot", repoSnapshotResponse.getSnapshots());
+        }, exception -> { throw new AssertionError(exception); }));
+
+        repoSnapshotRequest = new GetSnapshotsRequest().repository(repoName)
+            .snapshots(currSnapshotOnly)
+            .ignoreUnavailable(true)
+            .verbose(false);
+        getSnapshotsAction.execute(task, repoSnapshotRequest, ActionListener.wrap(repoSnapshotResponse -> {
+            assertNotNull("snapshots should be set as we are checking the current snapshot", repoSnapshotResponse.getSnapshots());
+        }, exception -> { throw new AssertionError(exception); }));
```

Comment:
why are you trying to call it multiple times here? if really needed it can be a for loop instead.

Reply:
This is to capture all the combinations of `ignoreUnavailable` and `verbose` attribute of the getSnapshotRequest call.